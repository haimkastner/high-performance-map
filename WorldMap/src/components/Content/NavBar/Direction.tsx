import React, { useState, useEffect } from "react";
import { Box, Button, TextField, Typography } from "@material-ui/core";
import {
  subsribeState,
  unsubsribeState,
  subscribeHookState
} from "../../../services/stats-store";
interface State {
  speed: string;
}

export function Direction() {
  const [speed, setSpeed] = useState(1);
  const [direction, setDirection] = useState(1);

  let token: any;
  useEffect(() => {
    token = subscribeHookState("direction", setDirection);
    token = subscribeHookState("speed", setSpeed);
  }, []);

  return (
    <Box>
      <Typography variant="h2" gutterBottom>
        function {direction} {speed}
      </Typography>
    </Box>
  );
}
