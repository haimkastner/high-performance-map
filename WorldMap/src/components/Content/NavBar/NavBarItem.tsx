import { createStyles, Grid, makeStyles, Theme, Typography } from '@material-ui/core';
import React from 'react';

interface Props {
  children?: JSX.Element;
  width: number | string;
  height: number | string;
  name: string;
  value: number | string;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    navItemInput: {
      backgroundColor: '#1D1D1D',
      borderColor: 'rgba(255, 255, 255, 0.5)',
      borderStyle: 'solid',
      borderWidth: 1,
      boxSizing: 'border-box',
      color: '#FFFFFF',
      fontFamily: 'Assistant',
      opacity: 0.6,
      textAlign: 'center',
    },
    navItemName: {
      color: '#FFFFFF',
      fontFamily: 'Assistant',
      fontSize: 16,
      fontWeight: 600,
    },
  }),
);

export function NavBarItem(props: Props) {
  const classes = useStyles();

  const navItemStyle = {
    height: props.height,
    width: props.width,
  };

  return (
    <Grid container direction="row" justify="space-around" alignItems="center" style={navItemStyle}>
      <Typography id="name" className={classes.navItemName}>
        {props.name}
      </Typography>
      <Typography id="value" className={classes.navItemInput} style={navItemStyle}>
        {props.value.toString()}
      </Typography>
    </Grid>
  );
}
