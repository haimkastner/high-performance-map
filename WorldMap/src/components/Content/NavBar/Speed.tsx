import React from "react";
import { Box, Button, TextField, Typography } from "@material-ui/core";
import { subscribeState, unsubscribeState, setSharedState } from "../../../services/stats-store";
interface State {
  speed: string;
}

export class Speed extends React.Component<any, State> {
  token: any;

  state: State = {
    speed: "0"
  };

  componentDidMount() {
    this.token = subscribeState("speed", this);

    setTimeout(() =>{
        setSharedState('speed', 457286278);
    }, 5000);
  }
  componentWillUnmount() {
    unsubscribeState(this.token);
  }
  render() {
    return (
      <Box>
        <Typography variant="h2" gutterBottom>
          class {this.state.speed}
        </Typography>
      </Box>
    );
  }
}
