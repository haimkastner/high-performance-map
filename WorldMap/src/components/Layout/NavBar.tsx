import React from 'react';
import styles from './Styles/NavBar.module.css';

export class NavBar extends React.Component {

    render() {
        return (<h1 className={styles.Downbar}>NavBar</h1>);
    }
}
