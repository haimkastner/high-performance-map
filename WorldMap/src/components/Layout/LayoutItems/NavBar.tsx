import React from "react";
import styles from "./Styles/NavBar.module.css";
import { Speed } from "../../Content/NavBar/Speed";
import { Direction } from "../../Content/NavBar/Direction";
import { Box } from "@material-ui/core";

export class NavBar extends React.Component<any, { speed: boolean }> {
  state: { speed: boolean } = {
    speed: false
  };

  componentDidMount() {
    setTimeout(() => {
      this.setState({
        speed: true
      });
    }, 4000);
  }

  render() {
    return (
      <Box>
        {this.state.speed ? <Direction></Direction> : ""}
        <Speed />
      </Box>
    );
  }
}
