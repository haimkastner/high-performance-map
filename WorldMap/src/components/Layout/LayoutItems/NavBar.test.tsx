import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { NavBar } from './NavBar';

describe('NavBar Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NavBar />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // tslint:disable-next-line: no-empty
  describe('NavBar Component Logic', () => {});
});
