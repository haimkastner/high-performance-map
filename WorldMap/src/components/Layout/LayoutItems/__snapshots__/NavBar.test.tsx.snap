// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component matches the snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row-reverse MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row-reverse MuiGrid-justify-xs-space-between"
    style={
      Object {
        "width": 242,
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      style={
        Object {
          "height": "42px",
          "width": 100,
        }
      }
    >
      <p
        className="MuiTypography-root makeStyles-navItemName-240 MuiTypography-body1"
        id="name"
      >
        כיוון ההפלגה
      </p>
      <p
        className="MuiTypography-root makeStyles-navItemInput-239 MuiTypography-body1"
        id="value"
        style={
          Object {
            "height": "42px",
            "width": 100,
          }
        }
      >
        NaN
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      style={
        Object {
          "height": "42px",
          "width": 132,
        }
      }
    >
      <p
        className="MuiTypography-root makeStyles-navItemName-240 MuiTypography-body1"
        id="name"
      >
        מהירות
      </p>
      <p
        className="MuiTypography-root makeStyles-navItemInput-239 MuiTypography-body1"
        id="value"
        style={
          Object {
            "height": "42px",
            "width": 132,
          }
        }
      >
        NaN
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row-reverse MuiGrid-justify-xs-space-between"
    style={
      Object {
        "width": 289,
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      style={
        Object {
          "height": "42px",
          "width": 120,
        }
      }
    >
      <p
        className="MuiTypography-root makeStyles-navItemName-240 MuiTypography-body1"
        id="name"
      >
        כיוון מספינתי
      </p>
      <p
        className="MuiTypography-root makeStyles-navItemInput-239 MuiTypography-body1"
        id="value"
        style={
          Object {
            "height": "42px",
            "width": 120,
          }
        }
      >
        -1
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      style={
        Object {
          "height": "42px",
          "width": 159,
        }
      }
    >
      <p
        className="MuiTypography-root makeStyles-navItemName-240 MuiTypography-body1"
        id="name"
      >
        טווח מספינתי
      </p>
      <p
        className="MuiTypography-root makeStyles-navItemInput-239 MuiTypography-body1"
        id="value"
        style={
          Object {
            "height": "42px",
            "width": 159,
          }
        }
      >
        -1
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row-reverse MuiGrid-justify-xs-space-between"
    style={
      Object {
        "width": 291,
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      style={
        Object {
          "height": "42px",
          "width": 161,
        }
      }
    >
      <p
        className="MuiTypography-root makeStyles-navItemName-240 MuiTypography-body1"
        id="name"
      >
        גובה/עומק
      </p>
      <p
        className="MuiTypography-root makeStyles-navItemInput-239 MuiTypography-body1"
        id="value"
        style={
          Object {
            "height": "42px",
            "width": 161,
          }
        }
      >
        NaN
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
      style={
        Object {
          "height": "42px",
          "width": 120,
        }
      }
    >
      <p
        className="MuiTypography-root makeStyles-navItemName-240 MuiTypography-body1"
        id="name"
      >
        הגדרות
      </p>
      <p
        className="MuiTypography-root makeStyles-navItemInput-239 MuiTypography-body1"
        id="value"
        style={
          Object {
            "height": "42px",
            "width": 120,
          }
        }
      >
        ?
      </p>
    </div>
  </div>
</div>
`;
